src/main/java/seedu/exercise/model/exercise/CustomProperty.java

```diff
@@ -0,0 +1,117 @@
+package seedu.exercise.model.exercise;
+
+import static seedu.exercise.commons.util.AppUtil.checkArgument;
+import static seedu.exercise.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.exercise.logic.parser.Prefix;
+
+/**
+ * Represents a custom property that an {@code Exercise} object can have.
+ */
+public class CustomProperty {
+    public static final String FULL_NAME_CONSTRAINTS = "Full names should contain only alphabets and should"
+        + " not be blank.";
+    public static final String SHORT_NAME_CONSTRAINTS = "Short names should contain only alphabets and should"
+        + " have no spaces and should not be blank.";
+
+    private final Prefix prefix;
+    private final String fullName;
+    private final ParameterType parameterType;
+
+    /**
+     * Instantiates a new {@code CustomProperty} instance.
+     *
+     * @param fullName      the full name of the custom property
+     * @param prefix        the short name (prefix) for the custom property
```

Comment:
Was thinking if we should just standardize the term `prefix` instead of switching between `shortName` and `prefix`. It is easier to follow the code based on the consistent naming of the variable. 

Taking it one step deeper, I the shortName prefix for the custom command can even be modified like this:
`custom p/PREFIX_NAME f/FULL_NAME t/PARAMETER_TYPE`
instead of the current.
`custom s/SHORT_NAME f/FULL_NAME p/PARAMETER_TYPE`

Compare `custom p/r f/Ratings t/Text` with `custom s/r f/Ratings p/Text`. The former feels more intuitive to me personally. But this is a quite subjective and is a drastic suggestion. Need to meet to discuss further.

Reply:
As of now, I have defined the following terms as such:
- Prefix: the short name of a parameter together with "/"
- Short name: the alias of a parameter 
As an example, the short name of the "muscle" parameter is "m" and its corresponding prefix is "m/". 

One advantage of distinguishing short name and prefix is that it allows easier validation of the new prefixes for the custom properties.

However, as you have mentioned, it might be confusing for others. 

We can see how it goes during the next meeting.